File: java/parser/src/main/java/gov/lbl/parser/parser/Short_class_specifierVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Array_subscripts;
import gov.lbl.parser.domain.Base_prefix;
import gov.lbl.parser.domain.Class_modification;
import gov.lbl.parser.domain.Comment;
import gov.lbl.parser.domain.Enum_list;
import gov.lbl.parser.domain.Name;
import gov.lbl.parser.domain.Short_class_specifier;
import gov.lbl.parser.domain.Short_class_specifier_value;

public class Short_class_specifierVisitor extends modelicaBaseVisitor<Short_class_specifier> {
    @Override
    public Short_class_specifier visitShort_class_specifier(modelicaParser.Short_class_specifierContext ctx) {
        String identifier = ctx.IDENT().getText();

        Base_prefixVisitor base_prefixVisitor = new Base_prefixVisitor();
        Base_prefix base_prefix = ctx.base_prefix() == null ? null : ctx.base_prefix().accept(base_prefixVisitor);
        Array_subscriptsVisitor array_subscriptsVisitor = new Array_subscriptsVisitor();
        Array_subscripts array_subscripts = ctx.array_subscripts() == null ? null : ctx.array_subscripts().accept(array_subscriptsVisitor);
        NameVisitor nameVisitor = new NameVisitor();
        Name name = ctx.name() == null ? null : ctx.name().accept(nameVisitor);
        Class_modificationVisitor class_modificationVisitor = new Class_modificationVisitor();
        Class_modification class_modification = ctx.class_modification() == null ? null : ctx.class_modification().accept(class_modificationVisitor);
        CommentVisitor commentVisitor = new CommentVisitor();
        Comment comment = ctx.comment() == null ? null : ctx.comment().accept(commentVisitor);
        Enum_listVisitor enum_listVisitor = new Enum_listVisitor();
        Enum_list enum_list = ctx.enum_list() == null ? null : ctx.enum_list().accept(enum_listVisitor);
        
        Short_class_specifier_value short_class_specifier_value = new Short_class_specifier_value(base_prefix, name, array_subscripts, class_modification, comment, enum_list);

        return new Short_class_specifier(identifier, short_class_specifier_value);
    }
}

File: jsParser/parser/Short_class_specifierVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Short_class_specifier = require('../domain/Short_class_specifier')
const Short_class_specifier_value = require('../domain/Short_class_specifier_value')

const Base_prefixVisitor = require('./Base_prefixVisitor')
const NameVisitor = require('./NameVisitor')
const Array_subscriptsVisitor = require('./Array_subscriptsVisitor')
const Class_modificationVisitor = require('./Class_modificationVisitor')
const CommentVisitor = require('./CommentVisitor')
const Enum_listVisitor = require('./Enum_listVisitor')

class Short_class_specifierVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitShort_class_specifier (ctx) {
    let identifier = ''
    var short_class_specifier_value = null
    if (ctx.IDENT()) {
      identifier = ctx.IDENT().getText()
    }

    let base_prefix = null
    let name = null
    let array_subscripts = null
    let class_modification = null
    let comment = null
    let enum_list = null

    if (ctx.base_prefix()) {
      const base_prefixVisitor = new Base_prefixVisitor.Base_prefixVisitor()
      base_prefix = base_prefixVisitor.visitBase_prefix(ctx.base_prefix())
    }
    if (ctx.name()) {
      const nameVisitor = new NameVisitor.NameVisitor()
      name = nameVisitor.visitName(ctx.name())
    }
    if (ctx.array_subscripts()) {
      const array_subscriptsVisitor = new Array_subscriptsVisitor.Array_subscriptsVisitor()
      array_subscripts = array_subscriptsVisitor.visitArray_subscripts(ctx.array_subscripts())
    }
    if (ctx.class_modification()) {
      const class_modificationVisitor = new Class_modificationVisitor.Class_modificationVisitor()
      class_modification = class_modificationVisitor.visitClass_modification(ctx.class_modification())
    }
    if (ctx.comment()) {
      const commentVisitor = new CommentVisitor.CommentVisitor()
      comment = commentVisitor.visitComment(ctx.comment())
    }
    if (ctx.enum_list()) {
      const enum_listVisitor = new Enum_listVisitor.Enum_listVisitor()
      enum_list = enum_listVisitor.visitEnum_list(ctx.enum_list())
    }
    var short_class_specifier_value = new Short_class_specifier_value.Short_class_specifier_value(base_prefix, name, array_subscripts, class_modification, comment, enum_list)

    return new Short_class_specifier.Short_class_specifier(identifier, short_class_specifier_value)
  }
};

Short_class_specifierVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitShort_class_specifier = this.visitShort_class_specifier
exports.Short_class_specifierVisitor = Short_class_specifierVisitor
